src/figuresClasses.ts

```diff
@@ -2,10 +1,84 @@
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Error: length of one side of a triangle is <= 0');
+    }
+
+    if (
+      this.a >= this.b + this.c ||
+      this.b >= this.a + this.c ||
+      this.c >= this.b + this.a
+    ) {
+      throw new Error(
+        `Error: sides ${this.a}, ${this.b} and ${this.c} can't form a triangle`,
+      );
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+
+    const s = (a + b + c) / 2;
+    const area = Math.sqrt(s * (s - a) * (s - b) * (s - c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  public shape: 'circle' = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('Error: radius must be greater than 0.');
+    }
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
+    const area = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  public shape: 'rectangle' = 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public width: number,
+    public height: number,
```

Comment:
In the `Rectangle` constructor, ensure you use `this` to access the properties `width` and `height` to correctly reference the instance variables.